modules/transport-netty4/src/internalClusterTest/java/org/opensearch/http/netty4/Netty4HeaderVerifierIT.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.http.netty4;
+
+import org.opensearch.OpenSearchNetty4IntegTestCase;
+import org.opensearch.core.common.transport.TransportAddress;
+import org.opensearch.http.HttpServerTransport;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase.ClusterScope;
+import org.opensearch.test.OpenSearchIntegTestCase.Scope;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.ReferenceCounted;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static io.netty.handler.codec.http.HttpHeaderNames.HOST;
+
+@ClusterScope(scope = Scope.TEST, supportsDedicatedMasters = false, numDataNodes = 1)
+public class Netty4HeaderVerifierIT extends OpenSearchNetty4IntegTestCase {
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(Netty4BlockingPlugin.class);
+    }
+
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/10260")
+    public void testThatNettyHttpServerRequestBlockedWithHeaderVerifier() throws Exception {
```

Comment:
@peternied I added back in the example header verifier and test that you added in https://github.com/opensearch-project/OpenSearch/pull/10261/commits/aec43e98f568088562b32c40f4f742f05c5ecff4. I see that the example headerVerifier calls `writeAndFlush` if there's a header called `blockme` in the HttpMessage, but otherwise lets other messages flow through the pipeline. This test case should be nearly identical to the one added in https://github.com/opensearch-project/OpenSearch/pull/10261/commits/aec43e98f568088562b32c40f4f742f05c5ecff4, but its failing with an exception instead of getting the expected response. I faced this issue when developing this PR where I tried numerous ways to call writeAndFlush, but there was always an exception where the HttpPipeliningHandler was receiving an object of type `LastHttpContent` (instead of a FullHttpRequest) and threw an error. 

Did you face the same error too? In the testing I performed, all the messages needed to flow through the pipeline, but the headerVerifier could set a channel attribute called `SHOULD_DECOMPRESS` that was passed to the decompressor to determine whether to decompress the HttpMessage or leave compressed and it would eventually get to the HttpPipeliningHandler as a FullHttpMessage. If there was an early response, the headerVerifier would also set a channel attribute called EARLY_RESPONSE. I updated the Netty4HttpRequestHandler to respond early if that channel attribute was set.